projects/ui-components/lf-metadata/lf-field-template-container/lf-field-template-container.component.html

```diff
@@ -1,15 +1,46 @@
 <div [hidden]="hideTemplate">
   <mat-form-field floatLabel="never" class="lf-field lf-template-picker lf-dropdown">
-    <mat-label class="lf-dropdown-label">{{emptyTemplateName | async}}</mat-label>
-    <mat-select name="template-dropdown" id="template-dropdown" class="lf-dropdown"
-        (selectionChange)="onTemplateChangedAsync($event)"
-        (openedChange)="onToggleDropdownAsync($event)"
-        [(value)]="templateSelected" [compareWith]="compareTemplateInfoFunc">
-        <mat-option selected>{{emptyTemplateName | async}}</mat-option>
-        <mat-option *ngFor="let template of availableTemplates" [value]="template">{{template.name}}
-        </mat-option>
+    <mat-label *ngIf="isDropdownLoading; else checkDropdownError" class="lf-dropdown-label">
+      <lf-loader-component></lf-loader-component>
+    </mat-label>
+    <ng-template #checkDropdownError>
+      <mat-label  *ngIf="isDropdownError; else showDefaultEmptyTemplate" class="lf-dropdown-label">
+        {{ emptyTemplateName | async }}
+      </mat-label>
+      <ng-template #showDefaultEmptyTemplate>
+        <mat-label class="lf-dropdown-label">
+          {{ emptyTemplateName | async }}
+        </mat-label>
+      </ng-template>
+    </ng-template>
+    <mat-select
+      name="template-dropdown"
+      id="template-dropdown"
+      class="lf-dropdown"
+      (selectionChange)="onTemplateChangedAsync($event)"
+      (openedChange)="onToggleDropdownAsync($event)"
+      [(value)]="templateSelected"
+      [compareWith]="compareTemplateInfoFunc"
+    >
+      <mat-option selected *ngIf="isDropdownDefault">{{ emptyTemplateName | async }}</mat-option>
+      <mat-option selected *ngIf="isDropdownError" class="dropdown-error-message">{{ AN_ERROR_OCCURED | async }}</mat-option>
+      <mat-option hidden="!isDropdownDefault" *ngFor="let template of availableTemplates" [value]="template"
+        >{{ template.name }}
+      </mat-option>
     </mat-select>
   </mat-form-field>
-  <div class="error-message" *ngIf="templateErrorMessage">{{templateErrorMessage | async}}</div>
-  <ng-template lfFieldView></ng-template>
+  <ng-container *ngIf="isTemplateDefault; else displayTemplateContainer"></ng-container>
```

Comment:
actually isTemplateDefault should not display anything (and this is to used when there is an error in the template selection)

Reply:
to clarify and document the logics (since the code is a little bit hard to read)
if isTemplateLoading
    show the loader
elseif isTemplateError
    show the error message
elseif isTemplateDisplay
   show lfFieldView
else (this is the isTemplateDefault state, the initial state, or when template has no fields)
   display nothing